src/Associations.cc

```diff
@@ -229,9 +213,7 @@ void Associations::collectRefStars(afw::table::SimpleCatalog &refCat, afw::geom:
         star->vxy = 0.;
 
         // Reject sources with non-finite fluxes and flux errors, and fluxErr=0 (which gives chi2=inf).
-        if (rejectBadFluxes &&
-            (!std::isfinite(defaultFlux) || !std::isfinite(defaultFluxErr) || defaultFluxErr == 0))
-            continue;
+        if (rejectBadFluxes && (!std::isfinite(flux) || !std::isfinite(fluxErr) || fluxErr == 0)) continue;
```

Comment:
I don't know of all the ways in which a catalog could have "bad" fluxes, but I would replace the last term with `fluxErr <= 0` to avoid anything with a (clearly wrong) negative flux error.